ecosystem/typescript/sdk_v2/src/transactions/instances/transactionPayload.ts

```diff
@@ -134,21 +136,52 @@ export class EntryFunction {
    * public entry fun transfer<CoinType>(from: &signer, to: address, amount: u64)
    * ```
    */
-  constructor(module_name: ModuleId, function_name: Identifier, type_args: Array<TypeTag>, args: Array<Uint8Array>) {
+  constructor(module_name: ModuleId, function_name: Identifier, type_args: Array<TypeTag>, args: Array<Serializable>) {
     this.module_name = module_name;
     this.function_name = function_name;
     this.type_args = type_args;
     this.args = args;
   }
 
+  /**
+   * A helper function to build a EntryFunction payload from raw primitive values
+   *
+   * @param module_name Fully qualified module name in format "AccountAddress::module_name" e.g. "0x1::coin"
+   * @param function_name Function name
+   * @param type_args Type arguments that move function requires.
+   *
+   * @example
+   * A coin transfer function has one type argument "CoinType".
+   * ```
+   * public(script) fun transfer<CoinType>(from: &signer, to: address, amount: u64,)
+   * ```
+   * @param args Arugments to the move function.
+   *
+   * @example
+   * A coin transfer function has three arugments "from", "to" and "amount".
+   * ```
+   * public(script) fun transfer<CoinType>(from: &signer, to: address, amount: u64,)
+   * ```
+   * @returns EntryFunction
+   */
+  static build(
+    module_name: `${string}::${string}`,
+    function_name: string,
+    type_args: Array<TypeTag>,
+    args: Array<Serializable>,
```

Comment:
In reference to the other comment below, this will eventually be:

```typescript
args: Array<EntryFunctionArgument>,

// where EntryFunctionArgument is:
export interface EntryFunctionArgument {
  /**
   * Serialize an argument to BCS-serialized bytes.
   */
  serialize(serializer: Serializer): void;

  /**
   * Serialize an argument as a type-agnostic, fixed byte sequence. The byte sequence contains
   * the number of the following bytes followed by the BCS-serialized bytes for a typed argument.
   */
  serializeForEntryFunction(serializer: Serializer): void;
}
```

we can talk about it tomorrow or something, it should make things a little more clear in terms of how we represent data for different types of payloads, and make it difficult/impossible to inadvertently use them incorrectly